sql/core/src/test/scala/org/apache/spark/sql/StatisticsSuite.scala

```diff
@@ -101,4 +101,47 @@ class StatisticsSuite extends QueryTest with SharedSQLContext {
       checkTableStats(tableName, expectedRowCount = Some(2))
     }
   }
+
+  test("test column-level statistics for data source table created in InMemoryCatalog") {
+    def checkColStats(colStats: BasicColStats, expectedColStats: BasicColStats): Unit = {
+      assert(colStats.dataType == expectedColStats.dataType)
+      assert(colStats.numNulls == expectedColStats.numNulls)
+      assert(colStats.max == expectedColStats.max)
+      assert(colStats.min == expectedColStats.min)
+      if (expectedColStats.ndv.isDefined) {
+        // ndv is an approximate value, so we just make sure we have the value
+        assert(colStats.ndv.get >= 0)
```

Comment:
It should be:

1. estimate column statistics for several times.
2. calculate mean and standard deviation of ndv for the multiple statistics.
3. check if the real ndv value is within (mean + 3 * sd, mean - 3 * sd).

Please correct me if it is wrong.